src/guides/v2.3/howdoi/checkout/checkout_custom_validation.md

```diff
@@ -0,0 +1,133 @@
+
+class LayoutProcessor
+{
+    /**
+    * @param \Magento\Checkout\Block\Checkout\LayoutProcessor $subject
+    * @param array $jsLayout
+    * @return array
+    */
+    public function afterProcess(
+    \Magento\Checkout\Block\Checkout\LayoutProcessor $subject,
+    array $jsLayout,
+    array $fields
+    ) {
+        // Firstname
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['firstname']['validation']['max_text_length'] = 20;
+
+        // Lastname
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['lastname']['validation']['max_text_length'] = 20;
+
+        // Company name
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['company']['validation']['max_text_length'] = 31;
+
+        // City
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['city']['validation']['max_text_length'] = 31;
+
+        // Region
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['region']['validation']['max_text_length'] = 21;
+
+        // Country
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['country_id']['validation']['max_text_length'] = 31;
+
+        // Postcode
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']['children']['postcode']['validation']['max_text_length'] = 13;
+
+        // Street address
+        $fields = $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']
+                ['shippingAddress']['children']['shipping-address-fieldset']
+                ['children']['street']['children'][0]['validation']['max_text_length'] = 40;
+
+        return $fields;
+    }
+}
+```
+
+## Step 4: Add validation for the billing address {#add-billing-validation}
+
+```php?start_inline=1
+$fields = $jsLayout['components']['checkout']['children']['steps']['children']['billing-step']['children']
+        ['shippingAddress']['children']['shipping-address-fieldset']['children']['firstname']['validation']['max_text_length'] = 20;
```

Comment:
Are all these references to the shipping address applicable for validating the billing address? Or is this doing something to ensure the values are the same for both types of addresses?